src/spec-configuration/containerFeaturesOrder.ts

```diff
@@ -3,133 +3,538 @@
+			const bCanonicalId = `${bResource}@${bDigest}`;
+
+			if (aCanonicalId === bCanonicalId && aOptions === bOptions) {
+				// Equal!
+				return 0;
+			}
+
+			// Sort by resource name
+			if (aResource !== bResource) {
+				return aResource.localeCompare(bResource);
+			}
+
+			const aTag = aSourceInfo.featureRef.tag;
+			const bTag = bSourceInfo.featureRef.tag;
+			// Sort by tags (if both have tags)
+			// Eg: 1.9.9, 2.0.0, 2.0.1, 3, latest
+			if ((aTag && bTag) && (aTag !== bTag)) {
+				return aTag.localeCompare(bTag);
+			}
+
+			// Sort by options
+			// TODO: Compares options Record by stringifying them.  This isn't fully correct.
+			if (aOptions !== bOptions) {
+				return aOptions.localeCompare(bOptions);
+			}
+
+			// Sort by manifest digest hash
+			if (aDigest !== bDigest) {
+				return aDigest.localeCompare(bDigest);
+			}
+
+			// Consider these two OCI Features equal.
+			return 0;
+
+		case 'direct-tarball':
+			throw new Error(`Feature type 'direct-tarball' will be supported but is unimplemented.`);
+
+		case 'file-path':
+			bSourceInfo = bSourceInfo as FilePathSourceInformation;
+			return aSourceInfo.resolvedFilePath.localeCompare(bSourceInfo.resolvedFilePath);
+
+		default:
+			throw new Error(`Feature dependencies are only supported in Features published (2) to an OCI registry, (2) as direct HTTPS tarball, or (3) as a local Feature.  Got type: '${aSourceInfo.type}'.`);
+	}
+}
+
+function featureSupportsDependencies(featureSet?: FeatureSet): boolean {
+	if (!featureSet) {
+		return false;
+	}
+
+	const publishType = featureSet.sourceInformation.type;
+	// TODO: Implement 'direct-tarball'.
+	return publishType === 'oci' /*|| publishType === 'direct-tarball'*/ || publishType === 'file-path';
+}
+
+async function applyOverrideFeatureInstallOrder(
+	params: CommonParams,
+	processFeature: (userFeature: DevContainerFeature) => Promise<FeatureSet | undefined>,
+	worklist: FNode[],
+	config: { overrideFeatureInstallOrder?: string[] },
+) {
+	const { output } = params;
+
+	if (!config.overrideFeatureInstallOrder) {
+		return worklist;
+	}
+
+	const overrideFeatureNodes: FNode[] = [];
+
+	// Create an artificial node for each Feature in the override property.
+	// Reverse iterate to remove out items from config that have been processed.
+	for (let j = config.overrideFeatureInstallOrder.length - 1; j >= 0; j--) {
+		const userFeatureId = config.overrideFeatureInstallOrder[j];
+		const overrideNode: FNode = {
+			type: 'artificial',
+			userFeatureId,
+			options: {},
+			dependsOn: [],
+			installsAfter: [],
+		};
+		const processed = await processFeature(overrideNode);
+		if (!processed) {
+			throw new Error(`Feature '${userFeatureId}' in 'overrideFeatureInstallOrder' could not be processed.`);
+		}
+
+		if (featureSupportsDependencies(processed)) {
+			overrideNode.featureSet = processed;
+			overrideFeatureNodes.push(overrideNode);
+			config.overrideFeatureInstallOrder.splice(j, 1);
```

Comment:
Why remove it?

Reply:
I remove it here so that later on when this object is used for legacyFeatures, we have spliced out all the overrides that have been processed in the dependencies algorithm.

Reply:
I would keep the original unchanged to keep it easier to understand in code and in the debugger. Can we modify a copy and pass that around to avoid modifying the original array?

Or better: Remove the legacy algorithm and let the new one handle everything.

Reply:
Ok, I will make that change.  I was hoping to reduce the complexity by not supporting deprecated concepts in the new algorithm.  This was the tradeoff, but perhaps now that i'm at this point it would not be too difficult. Perhaps that can be a later improvement - I would love to delete the previous code if it didn't involve a lot more work :D